telegram/ext/filters.py

```diff
@@ -372,1895 +363,1807 @@ def name(self, name: str) -> NoReturn:
+        """
+        Add one or more sender chats to the allowed chat ids.
+
+        Args:
+            chat_id(:obj:`int` | Tuple[:obj:`int`] | List[:obj:`int`], optional):
+                Which sender chat ID(s) to allow through.
+        """
+        return super().add_chat_ids(chat_id)
+
+    def remove_usernames(self, username: SLT[str]) -> None:
+        """
+        Remove one or more sender chats from allowed usernames.
+
+        Args:
+            username(:obj:`str` | Tuple[:obj:`str`] | List[:obj:`str`], optional):
+                Which sender chat username(s) to disallow through.
+                Leading ``'@'`` s in usernames will be discarded.
+        """
+        return super().remove_usernames(username)
+
+    def remove_chat_ids(self, chat_id: SLT[int]) -> None:
+        """
+        Remove one or more sender chats from allowed chat ids.
+
+        Args:
+            chat_id(:obj:`int` | Tuple[:obj:`int`] | List[:obj:`int`], optional):
+                Which sender chat ID(s) to disallow through.
+        """
+        return super().remove_chat_ids(chat_id)
+
+    class _SUPERGROUP(MessageFilter):
         __slots__ = ()
-        name = 'Filters.passport_data'
+        name = "filters.ChatType.SUPERGROUP"
 
         def filter(self, message: Message) -> bool:
-            return bool(message.passport_data)
-
-    passport_data = _PassportData()
-    """Messages that contain a :class:`telegram.PassportData`"""
+            if message.sender_chat:
+                return message.sender_chat.type == TGChat.SUPERGROUP
+            return False
 
-    class _Poll(MessageFilter):
+    class _CHANNEL(MessageFilter):
         __slots__ = ()
-        name = 'Filters.poll'
+        name = "filters.SenderChat.CHANNEL"
 
         def filter(self, message: Message) -> bool:
-            return bool(message.poll)
+            if message.sender_chat:
+                return message.sender_chat.type == TGChat.CHANNEL
+            return False
 
-    poll = _Poll()
-    """Messages that contain a :class:`telegram.Poll`."""
+    SUPER_GROUP = _SUPERGROUP()
+    """Messages whose sender chat is a super group."""
+    CHANNEL = _CHANNEL()
+    """Messages whose sender chat is a channel."""
 
-    class _Dice(_DiceEmoji):
-        __slots__ = ()
-        # pylint: disable=no-member
-        dice = _DiceEmoji(DiceEmoji.DICE, DiceEmoji.DICE.name.lower())
-        darts = _DiceEmoji(DiceEmoji.DARTS, DiceEmoji.DARTS.name.lower())
-        basketball = _DiceEmoji(DiceEmoji.BASKETBALL, DiceEmoji.BASKETBALL.name.lower())
-        football = _DiceEmoji(DiceEmoji.FOOTBALL, DiceEmoji.FOOTBALL.name.lower())
-        slot_machine = _DiceEmoji(DiceEmoji.SLOT_MACHINE, DiceEmoji.SLOT_MACHINE.name.lower())
-        bowling = _DiceEmoji(DiceEmoji.BOWLING, DiceEmoji.BOWLING.name.lower())
-
-    dice = _Dice()
+
+SENDER_CHAT = SenderChat(allow_empty=True)
+"""Shortcut for :class:`filters.SenderChat(allow_empty=True)`"""
```

Comment:
> Oh, so here also you want to do `filters.SenderChat.ALL` which is a shortcut for `filters.SenderChat(allow_empty=True)`?

That was my idea, yes. On second glance, this might not be easy to implement - a class containing an instance of itself. If it gives you too much troubles, let's keep it as is :)

> And the same for the other subclasses of `_ChatUserBaseFilter` or only this one since this offers `CHANNEL` and `SUPERGROUP`?

I'd say only here. `filters.{ViaBot, User, Chat}` don't have any filter-attributes, so

Reply:
yep, not possible to have a class containing an instance of itself, keeping as is.